File: clients/java3/common/src/org/openamq/framing/ContentBody.java
package org.openamq.framing;

import org.apache.mina.common.ByteBuffer;

/**
 * @author Robert Greig (robert.j.greig@jpmorgan.com)
 */
public class ContentBody extends AMQBody
{
    public static final byte TYPE = 3;

    /** TODO: consider whether this should be a pointer into the ByteBuffer to avoid copying */
    public byte[] payload;

    protected byte getType()
    {
        return TYPE;
    }

    public int getSize()
    {
        return (payload == null?0:payload.length);
    }

    public void writePayload(ByteBuffer buffer)
    {
        if (payload != null)
        {
            buffer.put(payload);
        }
    }

    protected void populateFromBuffer(ByteBuffer buffer, long size) throws AMQFrameDecodingException
    {
        if (size > 0)
        {
            payload = new byte[(int)size];
            buffer.get(payload);
        }
    }

    public static AMQFrame createAMQFrame(int channelId, ContentBody body)
    {
        final AMQFrame frame = new AMQFrame();
        frame.channel = channelId;
        frame.bodyFrame = body;
        return frame;
    }
}

File: dotnet/xmscommon/framing/ContentBody.cs
using System;
using jpmorgan.mina.common;
using log4net;

namespace OpenAMQ.Framing
{
    public class ContentBody : IBody
    {
        public static readonly byte TYPE = 3;

        /// <summary>
        /// 
        /// </summary>
        /// TODO: consider whether this should be a pointer into the ByteBuffer to avoid copying */
        public byte[] Payload;

        #region IBody Members

        public byte BodyType
        {
            get
            {
                return TYPE;
            }
        }

        public uint Size
        {
            get
            {
                return (ushort)(Payload == null ? 0 : Payload.Length);
            }
        }

        public void WritePayload(ByteBuffer buffer)
        {
            if (Payload != null)
            {
                buffer.Put(Payload);
            }
        }

        public void PopulateFromBuffer(ByteBuffer buffer, uint size)
        {
            if (size > 0)
            {
                Payload = new byte[size];
                buffer.Get(Payload);
            }
        }

        #endregion

        public static AMQFrame CreateAMQFrame(ushort channelId, ContentBody body)
        {
            AMQFrame frame = new AMQFrame();
            frame.Channel = channelId;
            frame.BodyFrame = body;
            return frame;
        }
    }
}
